scss/_base_typography-definitions.scss

```diff
@@ -244,42 +244,68 @@
   // Placeholder naming: $sp + -- + {preceding element} + {following element}, e.g. for p + h1: $sp--ph1
 
   %sp--ph1 {
-    padding-top: map-get($nudges, h1-mobile) + $spv--x-large;
+    padding-top: map-get($nudges, h1-mobile);
+    position: relative;
 
-    @media (min-width: $breakpoint-heading-threshold) {
-      padding-top: map-get($nudges, h1) + $spv--x-large;
+    &::before {
+      content: '';
+      display: block;
+      height: $sp-between-p-and-heading;
     }
```

Comment:
moving the padding to a pseudoelement would make `u-no-padding--top` useless and would be a breaking change in case anyone is using headings with `u-no-padding--top`.

We would need to modify the `u-no-padding--top` utility to also remove pseudoelement top padding.

Could we keep the `spv--x-large` on `padding-top`, rather than a pseudoelement, but keep the `$include-paragraph-heading-spacing` rule so that extending projects have the option to disable that spacing?